frontend/server/src/Controllers/User.php

```diff
@@ -2033,31 +2096,29 @@ public static function apiUpdate(\OmegaUp\Request $r): array {
         }
 
         $newSchoolId = $currentSchoolId;
-        if (!is_null($r['school_id'])) {
-            if (is_numeric($r['school_id'])) {
-                $school = \OmegaUp\DAO\Schools::getByPK($r['school_id']);
-                if (is_null($school)) {
-                    throw new \OmegaUp\Exceptions\InvalidParameterException(
-                        'parameterInvalid',
-                        'school'
-                    );
-                }
-                $newSchoolId = $school->school_id;
-            } elseif (empty($r['school_name'])) {
-                $newSchoolId = null;
-                $r['school_id'] = null;
-            } else {
-                $response = \OmegaUp\Controllers\School::apiCreate(new \OmegaUp\Request([
+        if (!is_null($r['school_id']) && $r['school_id'] !== '') {
+            $school = \OmegaUp\DAO\Schools::getByPK(intval($r['school_id']));
+            if (is_null($school)) {
+                throw new \OmegaUp\Exceptions\InvalidParameterException(
+                    'parameterInvalid',
+                    'school'
+                );
+            }
+            $newSchoolId = $school->school_id;
+        }
+        if (is_null($newSchoolId) && !empty($r['school_name'])) {
+            $response = \OmegaUp\Controllers\School::apiCreate(
+                new \OmegaUp\Request([
                     'name' => $r['school_name'],
                     'country_id' => !is_null(
                         $state
                     ) ? $state->country_id : null,
                     'state_id' => !is_null($state) ? $state->state_id : null,
                     'auth_token' => $r['auth_token'],
-                ]));
-                $r['school_id'] = $response['school_id'];
-                $newSchoolId = $response['school_id'];
-            }
+                ])
+            );
+            $r['school_id'] = $response['school_id'];
```

Comment:
será posible no modificar `$r`? se vale hacerlo en otro cambio.

Reply:
Se me ocurre crear `UserParams`, pero eso valdría la pena hacerlo en otro cambio como comentas.

Reply:
pero eso no debería ser necesario porque esto es un API y nadie _debería_ llamarlo. o hay alguien llamando este método que no debería?

onda, cambiar esto por

```suggestion
            $pickedSchoolId = $response['school_id'];
```

y usar `$pickedSchoolId` a partir de aquí en vdz de `$r['school_id']` debería bastar.

Reply:
Listo.

Parece que ya no es necesario utilizar `$pickedSchoolId` después de esto, porque la variable a usar es `$newSchoolId`. 

Ya ni siquiera era necesario agregarlo a `$r` para nada porque este valor se asigna fuera de la función `updateValueProperties`.